platform/macos/detect.py

```diff
@@ -86,27 +84,15 @@ def configure(env):
 
     ## Build type
 
-    if env["target"] == "release":
-        if env["optimize"] == "speed":  # optimize for speed (default)
-            env.Prepend(CCFLAGS=["-O3", "-fomit-frame-pointer", "-ftree-vectorize"])
-        elif env["optimize"] == "size":  # optimize for size
-            env.Prepend(CCFLAGS=["-Os", "-ftree-vectorize"])
+    # TODO: Re-evaluate the need for this / streamline with common config.
+    if env["target"] == "template-release":
+        if env["optimize"].startswith("speed"):
+            env.Prepend(CCFLAGS=["-fomit-frame-pointer", "-ftree-vectorize"])
```

Comment:
These options are not macOS or clang specific, so probably should be set everywhere (`/Oy` for MSVC).

With exception to `-no_deduplicate` (LD64 flag).

Reply:
Indeed, currently we seem to set `-fomit-frame-pointer -ftree-vectorize` only for Clang builds: Android, iOS, macOS.
I don't know why / if there's a problem using those with GCC. I guess we could try them out in a separate PR so that we don't have to deal with changing this optimization in this massive refactor.

`/Oy` for MSVC seems implied by both `/O1` and `/O2` so I guess it's not needed: https://learn.microsoft.com/en-us/cpp/build/reference/oy-frame-pointer-omission?view=msvc-170

Reply:
I researched this a bit in #66296, and found that we don't need these flags at all, so #66297 removes them.